### PR TITLE
Fixed matching of rules to specific projects

### LINKED ISSUE #182: spec.match.target does not work with projects
I Have multiple folders and projects under organization, I am trying to write policy constraints for each project individually.

I'm having issue similar to #134 

### My config is as below

Terraform Version
`Terraform v0.14.4`
`provider registry.terraform.io/hashicorp/google v3.56.0`

Terraform Validator version
`Build version: 2020-09-24`

Policy library from [Forseti Security](https://github.com/forseti-security/policy-library)

### My Issue:

```
match:
    target: ["organization/<org_id>/project/<project_number>"]
```

Does not seem to work. I always get a No Violation Found result.

However,

```
match:
    target: ["organization/<org_id>/project/*"]
```
works. Throws proper constraint violations.

when I run 
`terraform-validator validate --policy-path=${POLICY_PATH} ./terraform.tfplan.json --verbose=true`

I get the following 

```
2021/02/16 08:21:01 [INFO] Authenticating using DefaultClient...
2021/02/16 08:21:01 [INFO]   -- Scopes: [https://www.googleapis.com/auth/compute https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/ndev.clouddns.readwrite https://www.googleapis.com/auth/devstorage.full_control https://www.googleapis.com/auth/userinfo.email]
2021/02/16 08:21:01 [INFO] Instantiating GCE client for path https://www.googleapis.com/compute/v1/projects/
2021/02/16 08:21:01 [INFO] Instantiating GCE Beta client for path https://www.googleapis.com/compute/beta/projects/
2021/02/16 08:21:01 [INFO] Instantiating GKE client for path https://container.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating GKE Beta client for path https://container.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud DNS client for path https://www.googleapis.com/dns/v1/projects/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud DNS Beta client for path https://www.googleapis.com/dns/v1beta2/projects/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud KMS client for path https://cloudkms.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Stackdriver Logging client for path https://logging.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Storage client for path https://www.googleapis.com/storage/v1/
2021/02/16 08:21:01 [INFO] Instantiating Google SqlAdmin client for path https://sqladmin.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Pubsub client for path https://pubsub.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Dataflow client for path https://dataflow.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud ResourceManager client for path https://cloudresourcemanager.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud ResourceManager V client for path https://cloudresourcemanager.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud Runtimeconfig client for path https://runtimeconfig.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud IAM client for path https://iam.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud IAMCredentials client for path https://iamcredentials.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud Service Management client for path https://servicemanagement.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud Service Usage client for path https://serviceusage.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud Billing client for path https://cloudbilling.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud Build client for path https://cloudbuild.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud BigQuery client for path https://www.googleapis.com/bigquery/v2/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud CloudFunctions Client for path https://cloudfunctions.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud BigtableAdmin for path https://bigtableadmin.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud Source Repo client for path https://sourcerepo.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud Spanner client for path https://spanner.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud Dataproc client for path https://dataproc.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud Dataproc Beta client for path
2021/02/16 08:21:01 [INFO] Instantiating Filestore client for path https://file.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud IoT Core client for path https://cloudiot.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating App Engine client for path https://appengine.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Cloud Composer client for path https://composer.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Service Networking client for path https://servicenetworking.googleapis.com/
2021/02/16 08:21:01 [INFO] Instantiating Google Cloud Storage Transfer client for path https://storagetransfer.googleapis.com/
2021/02/16 08:21:03 [WARN] Failed to retrieve project_id for //cloudresourcemanager.googleapis.com/projects/<valid_project_id> from resource
2021/02/16 08:21:03 [WARN] Failed to retrieve project_id for //cloudresourcemanager.googleapis.com/projects/<valid_project_id> from resource
2021/02/16 08:21:03 [WARN] Failed to retrieve project_id for //cloudresourcemanager.googleapis.com/projects/<valid_project_id> from resource
2021/02/16 08:21:03 [WARN] Failed to retrieve project_id for //cloudresourcemanager.googleapis.com/projects/<valid_project_id> from resource
No violations found.
```

I am using the credentials of a service account to run, and the service account has the following roles at Organization level
```
Security Admin
Cloud Asset Viewer
```

### PR DESCRIPTION
**Pull request by melinath**
Fixed #182. 

Project Assets exported by CAI [use an identifier containing the project number](https://cloud.google.com/asset-inventory/docs/resource-name-format); however, terraform-google-conversion constructs the CAI identifier the project ID instead of the project number, because we don't have access to the project number until after it's created.

This PR updates (will update) to a version of terraform-google-conversion that uses the project number if available (while continuing to use project ID otherwise to properly match resources with each other) and adds a test to verify that this is behaving correctly.

### CODE REVIEW

**Comment by melinath** [comment]
The test failure only seems to happen on terraform 0.12 (not 0.13 or 0.14). I'm not clear on whether we need to keep supporting 0.12? I would prefer to but also have no idea what could be causing this behavior.

**Comment by joostschriek** [comment]
We have this issue and are still on tf 0.12. Support would be immensely appreciated.

**Comment by melinath** [comment]
@joostschriek thanks for the heads up :-)

**Review by rileykarson — APPROVED** [review]

File: test/cli_test.go [hunk 1]
```diff
@@ -288,7 +305,7 @@ func terraformInit(t *testing.T, executable, dir string) {
 }
 
 func terraformPlan(t *testing.T, executable, dir, tfplan string) {
-	terraformExec(t, executable, dir, "plan", "-input=false", "--out", tfplan)
+	terraformExec(t, executable, dir, "plan", "-input=false", "-refresh=false", "-out", tfplan)
```
Comment (melinath) [inline test/cli_test.go:308]: It looks like this is the change that's causing some IAM tests to fail on terraform 0.12. So the good news is that there's not a breaking change in the codebase, just in the tests. However, I will need to figure out a way to get the tests to pass on terraform 0.12 as well :-) This change is required in order to write update tests.

Comment (melinath) [reply test/cli_test.go:308]: It looks like the failing tests are all using an iam policy data source as an input for a different iam policy. Something in how that's handled (with respect to the -refresh option) must have changed between terraform 0.12 and 0.13. I don't think our tests need to use data sources; I'll remove them and enter the JSON values for the policies directly.

**Pull request merged**